src/pages/Search/SearchMoveTransactions.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {useMemo} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import {useSearchContext} from '@components/Search/SearchContext';
+import type {ListItem} from '@components/SelectionList/types';
+import {changeTransactionsReport} from '@libs/actions/Transaction';
+import Navigation from '@libs/Navigation/Navigation';
+import IOURequestEditReportCommon from '@pages/iou/request/step/IOURequestEditReportCommon';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Report} from '@src/types/onyx';
+
+type ReportListItem = ListItem & {
+    /** reportID of the report */
+    value: string;
+};
+
+function SearchMoveTransactions() {
+    const {selectedTransactions, clearSelectedTransactions} = useSearchContext();
+    const selectedTransactionsKeys = useMemo(() => Object.keys(selectedTransactions), [selectedTransactions]);
+
+    const [allReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {canBeMissing: false});
+    const transactionsReports = useMemo(() => {
+        const reports = Object.values(selectedTransactions).reduce((acc, transaction) => {
+            const report = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${transaction.reportID}`];
+            if (report) {
+                acc.add(report);
+            }
+            return acc;
+        }, new Set<Report>());
+        return [...reports];
```

Comment:
I think this condition is not quite right. We want to list all outstanding reports. We can use [getOutstandingReportsForUser](https://github.com/Expensify/App/blob/3e2724d2c1273c383ee9f7c7892ff91a6a0e6e51/src/libs/ReportUtils.ts#L9640) for that

Reply:
Here, I'm finding the reports to which the selected transactions belong. `getOutstandingReportsForUser` is used [here](https://github.com/software-mansion-labs/expensify-app-fork/blob/3e2724d2c1273c383ee9f7c7892ff91a6a0e6e51/src/pages/iou/request/step/IOURequestEditReportCommon.tsx#L53) to find reports that can be destination.